projects/angular/src/forms/common/abstract-container.ts

```diff
@@ -27,6 +27,7 @@ export abstract class ClrAbstractContainer implements DynamicWrapper, OnDestroy,
   @ContentChild(ClrControlHelper) controlHelperComponent: ClrControlHelper;
 
   control: NgControl;
+  secondaryControl: NgControl;
```

Comment:
Introducing a secondary control here is a bit controversial to me
1) we're adding case-specific code to a common parent. It will be only used in the date-range picker and nowhere else.
2) the concept of secondary control itself is a bit unusual. It leads to having a component with 2 roots. Why don't we add a common wrapper to the 2 inputs and add all range-specific code there?
3) it requires changes to many other core classes, which poses a high risk of regressions at seemingly unrelated components.

Reply:
Looks like the Container is tightly coupled with control element. As per existing code, only one control is allowed for each container. Making it 2 different containers(for primary and secondary control) will create 2 instances, and both controls cant talk to each other, which is an issue for date range picker. I agree that this will cause additional regression test for other components using the Abstract Container. But, we need to gracefully handle the second control scenario for all other components.